src/main/java/com/iota/iri/utils/thread/DedicatedScheduledExecutorService.java

```diff
@@ -0,0 +1,555 @@
+     * methods of the {@link ScheduledExecutorService} that do not throw {@link Exception}s when we try to start the
+     * same task multiple times. This is handy for implementing the "start" and "shutdown" methods of the background
+     * workers of IRI that would otherwise have to take care of not starting the same task more than once (when trying
+     * to be robust against coding errors or tests that start the same thread multiple times).<br />
+     * <br />
+     * <pre>
+     *     <code>Example:
+     *
+     *         private final static Logger logger = LoggerFactor.getLogger(MilestoneSolidifier.class);
+     *
+     *         private DedicatedScheduledExecutorService milestoneSolidifier = new DedicatedScheduledExecutorService(
+     *                 "Solidification Thread", logger, false);
+     *
+     *         // calling this multiple times will only start exactly one background job (ignore additional requests)
+     *         public void start() {
+     *             milestoneSolidifier.silentScheduleAtFixedRate(this::solidificationThread, 10, 50, MILLISECONDS);
+     *         }
+     *
+     *         // calling this multiple times will only stop the one instance that is running (if it is running)
+     *         public void shutdown() {
+     *             milestoneSolidifier.shutdownNow();
+     *         }
+     *
+     *         public void solidificationThread() {
+     *             System.out.println("I get executed every 50 milliseconds");
+     *         }
+     *     </code>
+     *     <code>Resulting Log Output:
+     *
+     *         [main] INFO  MilestoneSolidifier - Starting [Milestone Solidifier] (starts in 10ms / runs every 50ms) ...
+     *         [main] INFO  MilestoneSolidifier - [Milestone Solidifier] Started (execution #1) ...
+     *         [Milestone Solidifier] INFO  c.i.i.s.m.MilestoneSolidifier - I get executed every 50 milliseconds
+     *         [Milestone Solidifier] INFO  c.i.i.s.m.MilestoneSolidifier - I get executed every 50 milliseconds
+     *         [Milestone Solidifier] INFO  c.i.i.s.m.MilestoneSolidifier - I get executed every 50 milliseconds
+     *         [main] INFO  MilestoneSolidifier - Stopping [Milestone Solidifier] ...
+     *         [main] INFO  MilestoneSolidifier - [Milestone Solidifier] Stopped (after #4 executions) ...
+     *     </code
+     * </pre>
+     *
+     * @param threadName name of the thread (or null if we want to disable the automatic logging - exceptions will
+     *                   always be logged)
+     * @param logger logback logger that shall be used for the origin of the log messages
+     * @param debug debug flag that indicates if every "run" should be accompanied with a log message
+     */
+    public DedicatedScheduledExecutorService(String threadName, Logger logger, boolean debug) {
+        super(1);
+
+        this.threadName = threadName;
+        this.logger = logger;
+        this.debug = debug;
```

Comment:
Instead of having a separate debug flag can we maybe use `logger.isDebug()` throughout the class logic?

Reply:
How do we control the debug flag?
If it is by manually changing in the code it is fine for the dev who works on it, but not if we communicate with a user experiencing problems.
I would think this should be able to activate at least with `logger.isTrace()` so one can see the messages without changing the code.
Would you agree with that?

Reply:
It's only for the developer to be able to add extra log messages while developing a new task which allows him to determine if something obvious doesn't work as planned (like the given example of a task not terminating). This should not be controlled by the user.

Exceptions or problems that might arise within the scope of a user interacting with IRI should always be handled by the specific logic of the task. Enabling a system wide debug flag for "all" "Threads" in IRI and printing additional "start / stop" messages for all Threads that use this ExecutorService would in fact make it even harder to debug problems because the console would be spammed with completely useless messages that are not related to the problem (depending on the chosen intervals - every few milliseconds).

Reply:
This is why I offered to do in `trace` mode.
I understand your reasoning though and accept it.